cantor-s3/src/main/java/com/salesforce/cantor/s3/utils/S3Utils.java

```diff
@@ -0,0 +1,183 @@
+    private static final Logger logger = LoggerFactory.getLogger(ObjectsOnS3.class);
+
+    // read objects in 4MB chunks
+    private static final int streamingChunkSize = 4 * 1024 * 1024;
+
+    public static void createBucket(final AmazonS3 s3Client, final String bucketName) throws IOException {
+        if (s3Client.doesBucketExistV2(bucketName)) {
+            logger.info("bucket '{}' already exists; ignoring create", bucketName);
+            return;
+        }
+        logger.info("bucket '{}' doesn't exist; creating it", bucketName);
+        s3Client.createBucket(bucketName);
+
+        // check bucket created successfully
+        if (!s3Client.doesBucketExistV2(bucketName)) {
+            throw new IOException("failed to create namespace on s3 with bucket name: " + bucketName);
+        }
+    }
+
+    public static void deleteBucket(final AmazonS3 s3Client, final String bucketName) throws IOException {
+        deleteObjects(s3Client, bucketName, null);
+
+        // now delete the bucket
+        s3Client.deleteBucket(bucketName);
+
+        // check bucket deleted successfully
+        if (s3Client.doesBucketExistV2(bucketName)) {
+            throw new IOException("failed to drop namespace on s3 with bucket name: " + bucketName);
+        }
+    }
+
+    public static Collection<String> getKeys(final AmazonS3 s3Client,
+                                             final String key,
+                                             final String prefix,
+                                             final int start,
+                                             final int count) throws IOException {
+        if (!s3Client.doesBucketExistV2(key)) {
+            throw new IOException(String.format("couldn't find bucket '%s'", key));
+        }
+
+        final Set<String> keys = new HashSet<>();
+        int index = 0;
+        ObjectListing listing = s3Client.listObjects(key, prefix);
+        do {
+            for (final S3ObjectSummary summary : listing.getObjectSummaries()) {
+                if (index < start) {
+                    logger.debug("skipping {} at index={} start={}", summary.getKey(), index++, start);
+                    continue;
+                }
+                keys.add(summary.getKey());
+
+                if (keys.size() == count) {
+                    logger.debug("retrieved {}/{} keys, returning early", keys.size(), count);
+                    return keys;
+                }
+            }
+
+            logger.debug("got {} keys from {}", listing.getObjectSummaries().size(), listing);
+            listing = s3Client.listNextBatchOfObjects(listing);
+        } while (listing.isTruncated());
+
+        return keys;
+    }
+
+    public static byte[] getObjectBytes(final AmazonS3 s3Client,
+                                        final String bucketName,
+                                        final String key) throws IOException {
+        if (!s3Client.doesObjectExist(bucketName, key)) {
```

Comment:
@AlexanderKouthoofd why do we need to first check if it exists, then do a get call? can't we just get and if it's null then it doesn't exist?